### PR TITLE
Fix zenodo export relative path

### LINKED ISSUE #617: Invalid path when exporting dataset to Zenodo
I added data to an existing dataset:
`jen@jen:~/projects/ace_data_management/renku-projects/meteorology/data$ renku dataset add meteorology /home/jen/projects/ace_data_management/data_to_archive_post_cruise/met_data/raw/data_file_header.txt`

then exported it to Zenodo: 

`jen@jen:~/projects/ace_data_management/renku-projects/meteorology/data$ renku dataset export meteorology zenodo`

After adding the access token, it seems that an invalid path to the files being exported, was used: 

` '/home/jen/projects/ace_data_management/renku-projects/meteorology/data/data/meteorology/README.txt'`

Please see below for the bug report: 

jen@jen:~/projects/ace_data_management/renku-projects/meteorology/data$ renku dataset export meteorology zenodo
Before exporting, you must configure an access token
Create one at: https://zenodo.org/account/settings/applications/tokens/new/
Access token: ###
  0%|                                                                                                                                   | 0/2 [00:00<?, ?it/s]
Ahhhhhhhh! You have found a bug. 🐞

1. Open an issue by typing "open";
2. Print human-readable information by typing "print";
3. See the full traceback without submitting details (default: "ignore").

Please select an action by typing its name (open, print, ignore) [ignore]: print
## Describe the bug
A clear and concise description.

## Details
*Please verify and redact the details.*

**Renku version:** 0.5.2
**OS:** Linux (#1 SMP Debian 4.9.144-3.1 (2019-02-19))
**Python:** 3.7.3

### Traceback

```
Traceback (most recent call last):
  File "[...]/site-packages/renku/cli/_exc.py", line 100, in main
    result = super().main(*args, **kwargs)
  File "[...]/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "[...]/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "[...]/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "[...]/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "[...]/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "[...]/site-packages/renku/cli/_client.py", line 83, in new_func
    result = ctx.invoke(method, client, *args, **kwargs)
  File "[...]/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "[...]/site-packages/renku/cli/dataset.py", line 495, in export_
    destination = exporter.export(publish)
  File "[...]/site-packages/renku/cli/_providers/zenodo.py", line 446, in export
    deposition.upload_file(file_.full_path, )
  File "[...]/site-packages/renku/cli/_providers/zenodo.py", line 349, in upload_file
    file = {'file': open(str(filepath), 'rb')}
FileNotFoundError: [Errno 2] No such file or directory: '/home/jen/projects/ace_data_management/renku-projects/meteorology/data/data/meteorology/README.txt'
```
Please let me know if you need any more information.

### PR DESCRIPTION
**Pull request by Panaetius**
# Description

Sets the `client` property of files in a dataset after the dataset has been loaded. This is needed to ensure the `full_path` property of files returns the correct path to a file irrespective of the current working directory.

Also adds a new unit test to check for this case

Fixes #617

## Type of change

Please select relevant options.

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

**Do not create pull request unless you checked all points.**

- [X] I have read and understood the **CONTRIBUTING** document.
- [X] I have performed a self-review of my own code.
- [X] I have signed and sent the [CLA document]
      (https://github.com/SwissDataScienceCenter/documentation/wiki/files/SDSC_Contributor_License_Agreement_v1.0.pdf).
- [X] I have added tests to cover my changes.
- [X] I have **NOT** removed any existing tests.
- [X] I have updated the documentation accordingly.
- [X] I have described *all* changes in the **CHANGELOG.rst**.
- [X] I have run ``./run-tests.sh`` locally.

### CODE REVIEW

File: renku/models/datasets.py [hunk 1]
```diff
@@ -465,3 +465,18 @@ def __attrs_post_init__(self):
...
+            for datasetfile in self.files:
+                if datasetfile.client is None:
+                    try:
+                        client, _, _ = self.client.resolve_in_submodules(
+                            self.client.find_previous_commit(
+                                datasetfile.path, revision='HEAD'
+                            ),
+                            datasetfile.path,
+                        )
+                    except KeyError:
```
Comment (rokroskar) [inline renku/models/datasets.py]: doesn't `KeyError` indicate that the file doesn't exist in the tree? in that case something is wrong - shouldn't we fail and let the user know in that case?

Comment (Panaetius) [reply renku/models/datasets.py]: `KeyError` is when the file wasn't found in `find_previous_commit`. There are a few unit tests that create a dataset with the `-t` parameter, which triggers the `KeyError` when the dataset is accessed unless it's handled this way. For instance `test_move_symlinks` has that problem.

Or are the unit tests broken and need to be adjusted?

Comment (rokroskar) [reply renku/models/datasets.py]: hmm... not sure - the `-t` flag allows you to specify a specific path within a git project. It's very possible that code does something funny and breaks the path in the dataset.

**Comment by Panaetius** [comment]
Everything should work now without the key error. There was an issue with `path`'s of `Datasetfile`'s not being set correctly for files in external git repos.

**Pull request merged**